### PR TITLE
add permissive movement to allowed

### PR DESCRIPTION
**Pull request by adev4a**
# PR Details
This PR adds Permissive movements to the lci strategic plugin planning logic. It extends the functionality for protected movements to permissive movements as well.
## Description
During voices integration testing it was found that SPAT messages published by the virtual econolite controller were being published with the permissive movement type.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lci_strategic_plugin/src/lci_strategic_plugin.cpp [hunk 1]
```diff
@@ -325,10 +325,12 @@ boost::optional<bool> LCIStrategicPlugin::canArriveAtGreenWithCertainty(const rc
     bool can_make_late_arrival = true;
 
     if (check_early)
-      can_make_early_arrival = (early_arrival_state_by_algo_optional.get().second == lanelet::CarmaTrafficSignalState::PROTECTED_MOVEMENT_ALLOWED);
+      can_make_early_arrival = ((early_arrival_state_by_algo_optional.get().second == lanelet::CarmaTrafficSignalState::PROTECTED_MOVEMENT_ALLOWED)
+      || (early_arrival_state_by_algo_optional.get().second == lanelet::CarmaTrafficSignalState::PERMISSIVE_MOVEMENT_ALLOWED));
```
Comment (MishkaMN) [inline lci_strategic_plugin/src/lci_strategic_plugin.cpp:329]: All of the OR or AND conditions added in this PR can be made concise by using helper function called 
`isAllowedMovement`  where 
```
bool isAllowedMovement(const lanelet::CarmaTrafficSignalState& state) {
    return state == lanelet::CarmaTrafficSignalState::PROTECTED_MOVEMENT_ALLOWED ||
           state == lanelet::CarmaTrafficSignalState::PERMISSIVE_MOVEMENT_ALLOWED;
}
```

**Review by MishkaMN — CHANGES_REQUESTED** [review]

File: lci_strategic_plugin/src/lci_strategic_plugin_algo.cpp [hunk 2]
```diff
@@ -369,15 +369,20 @@ std::tuple<rclcpp::Time, bool, bool> LCIStrategicPlugin::get_final_entry_time_an
         RCLCPP_DEBUG_STREAM(rclcpp::get_logger("lci_strategic_plugin"), "normal_arrival_signal_end_time: " << std::to_string(lanelet::time::toSec(normal_arrival_state_green_et_optional.get().first)));
 
         // nearest_green_signal_start_time = normal_arrival_signal_end_time (green guaranteed) - green_signal_duration
-        nearest_green_signal_start_time = rclcpp::Time(lanelet::time::toSec(normal_arrival_state_green_et_optional.get().first - traffic_light->signal_durations[lanelet::CarmaTrafficSignalState::PROTECTED_MOVEMENT_ALLOWED]) * 1e9);
+        if(traffic_light->signal_durations[lanelet::CarmaTrafficSignalState::PROTECTED_MOVEMENT_ALLOWED]){
```
[the remark this answers is not in the document]
Comment (adev4a) [reply lci_strategic_plugin/src/lci_strategic_plugin_algo.cpp]: Thats a nice suggestion

[the remark this answers is not in the document]
Comment (adev4a) [reply lci_strategic_plugin/src/lci_strategic_plugin_algo.cpp]: Just updated it

File: lci_strategic_plugin/src/lci_strategic_plugin_algo.cpp [hunk 3]
```diff
@@ -251,6 +250,17 @@ rclcpp::Duration LCIStrategicPlugin::get_earliest_entry_time(double remaining_di
 
 }
 
+int LCIStrategicPlugin::getClearanceDuration(lanelet::CarmaTrafficSignalPtr traffic_light)
```
Comment (MishkaMN) [inline lci_strategic_plugin/src/lci_strategic_plugin_algo.cpp:253]: The name is little misleading as it is getting movement allowed durations not clearance

**Review by MishkaMN — CHANGES_REQUESTED** [review]

**Comment by MishkaMN** [comment]
@adev4a is this PR still needed?

**Comment by adev4a** [comment]
Yes,  this PR wasn't merged since it hasn't been integration tested. But since we have no integration testing planned for this plugin currently, lets merge it and test when work begins on this plugin again.

**Review by MishkaMN — DISMISSED** [review]

File: lci_strategic_plugin/src/lci_strategic_plugin.cpp [hunk 4]
```diff
@@ -338,8 +338,8 @@ boost::optional<bool> LCIStrategicPlugin::canArriveAtGreenWithCertainty(const rc
 
 }
 
-bool LCIStrategicPlugin::isStateAllowedGreen(const lanelet::CarmaTrafficSignalState& state) {
-    return state == lanelet::CarmaTrafficSignalState::PROTECTED_MOVEMENT_ALLOWED ||
+bool LCIStrategicPlugin::isStateAllowedGreen(const lanelet::CarmaTrafficSignalState& state) const{
+    return state == lanelet::CarmaTrafficSignalState::PROTECTED_MOVEMENT_ALLOWED |
```
Comment (MishkaMN) [inline lci_strategic_plugin/src/lci_strategic_plugin.cpp:342]: aren't we supposed to use || operator?

**Review by MishkaMN — CHANGES_REQUESTED** [review]

**Review by MishkaMN — APPROVED** [review]

**Pull request merged**